### DISCUSSION TITLE
Configure Keycloak in React JS app

### DISCUSSION THREAD

**Discussion by sapiens-shahbaz-khan** [keycloak/keycloak#13120] [Q&A]
Hi,

My keycloak client is configured with access type : confidential and I have generated the secret key.
In my react JS app I am using the below provider
`import { ReactKeycloakProvider} from '@react-keycloak/web';`
In the provider I am passing **authClient & initOptions** as props which are as follows:
```
{
authClient: Keycloak({ clientId: 'secured-client', realm: 'the-realm', url: REACT_APP_KEYCLOAK_AUTH_DOMAIN, }), initOptions:{ onLoad: 'check-sso', messageReceiveTimeout: 2000, 
}
```
If I launch my app with keycloak server configured with access type confidential, it gives 401 error :
`error: "unauthorized_client" error_description: "Client secret not provided in request"`

Taking reference of the keycloak docs, how the java adapters can be configured,
I tried putting credentials in the **initOptions** and also **clientSecret** with the key. But it doesn't seem to work.

What is the right way to pass the secret key from a frontend application in my case a React JS app?

**Comment by gabrielqueirozdev** [comment]
Hello, do you fixed this bug?

**Reply by sapiens-shahbaz-khan** [reply]
You may refer to [discussions/11409]( https://github.com/keycloak/keycloak/discussions/11409). I had to do a customized solution, and mentioned in this ticket. Ty!
